google-cloud-firestore/src/main/java/com/google/cloud/firestore/BatchWriteResult.java

```diff
@@ -25,26 +25,53 @@
  * BatchWriteRequests.
  */
 public final class BatchWriteResult {
+  private final String key;
   @Nullable private final Timestamp writeTime;
   private final Status status;
   private final String message;
+  @Nullable private final Exception exception;
 
-  BatchWriteResult(@Nullable Timestamp timestamp, Status status, String message) {
+  BatchWriteResult(String key, @Nullable Timestamp timestamp, Status status, String message) {
+    this.key = key;
     this.writeTime = timestamp;
     this.status = status;
     this.message = message;
+    this.exception = null;
+  }
+
+  BatchWriteResult(
+      String key,
+      @Nullable Timestamp timestamp,
+      Status status,
+      String message,
+      Exception exception) {
```

Comment:
We can probably drop `status`, `message` and just take a FirestoreException if we update bulkCommit to only ever throw FirestoreExceptions.